### ISSUE TITLE
Calling Upscale from Colab is throwing an error

### ISSUE THREAD

**Issue by spuliz** [jina-ai/dalle-flow#28]
I tried upscaling my images but they always stay at 250x250 and not at 1024x1024. 
Server side error:
```
ERROR  upscaler/rep-0@42                                 
       FileNotFoundError('`../SwinIR/results/swinir_real_sr…                    
       is not a URL or a valid local path')  
```

```
    FileNotFoundError:                                                       
       `../SwinIR/results/swinir_real_sr_x4_large/ac3266c0f…                    
       is not a URL or a valid local path   
```

**Comment by JoanFM** [comment]
Hey spuliz, the error is clear, the path you are passing is not well seen by the Executor, I would maybe try sending the absolute path first of all

**Comment by spuliz** [comment]
╭──────── Traceback (most recent call last) ────────╮                    
       │ /usr/local/lib/python3.8/dist-packages/jina/serv… │                    
       │ in process_data                                   │                    
       │                                                   │                    
       │   153 │   │   │   │   if self.logger.debug_enable │                    
       │   154 │   │   │   │   │   self._log_data_request( │                    
       │   155 │   │   │   │                               │                    
       │ ❱ 156 │   │   │   │   return await self._data_req │                    
       │   157 │   │   │   except (RuntimeError, Exception │                    
       │   158 │   │   │   │   self.logger.error(          │                    
       │   159 │   │   │   │   │   f'{ex!r}'               │                    
       │                                                   │                    
       │ /usr/local/lib/python3.8/dist-packages/jina/serv… │                    
       │ in handle                                         │                    
       │                                                   │                    
       │   146 │   │   )                                   │                    
       │   147 │   │                                       │                    
       │   148 │   │   # executor logic                    │                    
       │ ❱ 149 │   │   return_data = await self._executor. │                    
       │   150 │   │   │   req_endpoint=requests[0].header │                    
       │   151 │   │   │   docs=docs,                      │                    
       │   152 │   │   │   parameters=params,              │                    
       │                                                   │                    
       │ /usr/local/lib/python3.8/dist-packages/jina/serv… │                    
       │ in __acall__                                      │                    
       │                                                   │                    
       │   251 │   │   # noqa: DAR201                      │                    
       │   252 │   │   """                                 │                    
       │   253 │   │   if req_endpoint in self.requests:   │                    
       │ ❱ 254 │   │   │   return await self.__acall_endpo │                    
       │   255 │   │   elif __default_endpoint__ in self.r │                    
       │   256 │   │   │   return await self.__acall_endpo │                    
       │   257                                             │                    
       │                                                   │                    
       │ /usr/local/lib/python3.8/dist-packages/jina/serv… │                    
       │ in __acall_endpoint__                             │                    
       │                                                   │                    
       │   272 │   │                                       │                    
       │   273 │   │   with _summary:                      │                    
       │   274 │   │   │   if iscoroutinefunction(func):   │                    
       │ ❱ 275 │   │   │   │   return await func(self, **k │                    
       │   276 │   │   │   else:                           │                    
       │   277 │   │   │   │   return func(self, **kwargs) │                    
       │   278                                             │                    
       │                                                   │                    
       │ /usr/local/lib/python3.8/dist-packages/jina/serv… │                    
       │ in arg_wrapper                                    │                    
       │                                                   │                    
       │   106 │   │   │   │   async def arg_wrapper(      │                    
       │   107 │   │   │   │   │   executor_instance, *arg │                    
       │   108 │   │   │   │   ):  # we need to get the su │                    
       │       the self                                    │                    
       │ ❱ 109 │   │   │   │   │   return await fn(executo │                    
       │   110 │   │   │   │                               │                    
       │   111 │   │   │   │   self.fn = arg_wrapper       │                    
       │   112 │   │   │   else:                           │                    
       │                                                   │                    
       │ /dalle/dalle-flow/executors/swinir/executor.py:62 │                    
       │ in upscale                                        │                    
       │                                                   │                    
       │   59 │   │   for d in docs:                       │                    
       │   60 │   │   │   if not d.tags.get('upscaled'):   │                    
       │   61 │   │   │   │   # only upscale once          │                    
       │ ❱ 62 │   │   │   │   self._upscale(d)             │                    
       │   63 │   │   │   │   d.blob = None                │                    
       │   64 │   │   │   │   d.embedding = None           │                    
       │   65                                              │                    
       │                                                   │                    
       │ /dalle/dalle-flow/executors/swinir/executor.py:42 │                    
       │ in _upscale                                       │                    
       │                                                   │                    
       │   39 │   │   │   subprocess.getoutput(f'python ma │                    
       │   40 │   │   )                                    │                    
       │   41 │   │   d.uri = os.path.join(self.output_pat │                    
       │ ❱ 42 │   │   d.convert_uri_to_datauri()           │                    
       │   43 │   │   d.tags['upscaled'] = True            │                    
       │   44 │   │   d.tags.update(kw)                    │                    
       │   45                                              │                    
       │                                                   │                    
       │ /usr/local/lib/python3.8/dist-packages/docarray/… │                    
       │ in convert_uri_to_datauri                         │                    
       │                                                   │                    
       │   46 │   │   :return: itself after processed      │                    
       │   47 │   │   """                                  │                    
       │   48 │   │   if not _is_datauri(self.uri):        │                    
       │ ❱ 49 │   │   │   blob = _uri_to_blob(self.uri)    │                    
       │   50 │   │   │   self.uri = _to_datauri(self.mime │                    
       │   51 │   │   return self                          │                    
       │   52                                              │                    
       │                                                   │                    
       │ /usr/local/lib/python3.8/dist-packages/docarray/… │                    
       │ in _uri_to_blob                                   │                    
       │                                                   │                    
       │   19 │   │   with open(uri, 'rb') as fp:          │                    
       │   20 │   │   │   return fp.read()                 │                    
       │   21 │   else:                                    │                    
       │ ❱ 22 │   │   raise FileNotFoundError(f'`{uri}` is │                    
       │   23                                              │                    
       │   24                                              │                    
       │   25 def _get_file_context(file):                 │                    
       ╰───────────────────────────────────────────────────╯                    
       FileNotFoundError:                                                       
       `../SwinIR/results/swinir_real_sr_x4_large/ac3266c0f…                    
       is not a URL or a valid local path

**Comment by spuliz** [comment]
Thanks @JoanFM - how and where should I add the absolute path in the docker files?

**Comment by JoanFM** [comment]
In the `flow.yml` you can check this:

```  - name: upscaler
    uses: SwinIRUpscaler
    py_modules:
      - executors/swinir/executor.py
    uses_with:
      swinir_path: ../SwinIR
```

Maybe u want to edit the `swinir_path` argument to point to where the folder will be located inside the docker image

**Comment by spuliz** [comment]
Hi @JoanFM are you able to repro the issue? More people are experiencing the same issue.
Steps to repro:
1) Install docker (following installation steps included in the readme of the repository)
2) Run docker
3) Open the Client provided on Google Colab and run all cells
4) See that image does not get upscaled but error is thrown.

**Comment by JoanFM** [comment]
I did not have time to reproduce, but can u make sure by logging inside the container, that the path `../SwinIR` does exist?

**Comment by spuliz** [comment]
Hi @JoanFM the path inside the docker is `../swinir` I modified flow.yml but is it not working still..

**Comment by JoanFM** [comment]
can u show what the current error is now? it seems it is trying to `chdir` but not possible, but what's exactly the error?

**Comment by spuliz** [comment]
@JoanFM 
<img width="393" alt="image" src="https://user-images.githubusercontent.com/15948985/171234024-32f49f21-ee7c-44c9-a33e-f189c78a178d.png">

**Comment by Dessix** [comment]
An absolute path of `/dalle/SwinIR` did not resolve the problem; it appears to be that the real issue is that the path itself is actually valid but absent- because, above that, `main_test_swinir.py` applies the SwinIR model-zoo model and OOMs with `RuntimeError: CUDA error: out of memory`. I've reproduced the issue by running via Docker Desktop on Windows 10 with an RTX 3090 (24GiB VRAM) on Cuda 11.6.0.

Upscaling does not appear to be viable without this path, and I have yet to find a workaround, though I'm considering stripping the other services temporarily.

**Comment by hanxiao** [comment]
Did you try building docker and run it via docker container? I just rebuild and run without any issue.

https://github.com/jina-ai/dalle-flow#run-in-docker

```bash
git clone https://github.com/jina-ai/dalle-flow.git
cd dalle-flow

docker build --build-arg GROUP_ID=$(id -g ${USER}) --build-arg USER_ID=$(id -u ${USER}) -t jinaai/dalle-flow .

docker run -p 51005:51005 -v $HOME/.cache:/home/dalle/.cache --gpus all jinaai/dalle-flow
```

**Comment by Dessix** [comment]
That's the way I built it in the first place- yes. The problem is that it allocates more memory than is available even on a 24GiB GPU, despite claiming to run in 21.7GiB.

The configuration being in `platform` memory mode would indicate that it should deallocate the memory for the other steps when it's done with them, but it seems to require enough of a baseline that the ~2 gigabytes used for upscaling is enough to pop it.

**Comment by ryakr** [comment]
I too am running into this issue on a 3090 running the code via the Docker image, but my error doesnt look to be related to a normal OOM. My system is reporting 4GiB, at least, still free in the VRAM when this occurs.

`RuntimeError: CUDA error: too many resources requested for launch `

It can usually upscale one or two images just fine before throwing this error until it is restarted again.
Here is the full output I get with that :
![image](https://user-images.githubusercontent.com/10369118/173271438-bf6c2e9a-a062-4309-8683-81899e3a59ad.png)

**Comment by Dessix** [comment]
@ryakr "CUDA error: Too many resources requested" implies that it is requesting too many threads per block. I'm not clear what may be different between our versions, but I am running a 3090 FTW3 rather than one based on the stock NVidia layout, so we may have actual hardware capability differences.

Worth noting: We're both hitting resource limits, just with different resources- and neither of them are handled by the runtime- which only encounters the issue when it realizes that the file at the expected path has not been created.

**Comment by ryakr** [comment]
@Dessix Yeah I figured our issues were related in some way, just thought I would drop in what i'm seeing to hopefully help narrow it down some. I'm running a 3090 Founders Edition myself, so its is exactly the stock NVIDIA layout.

In my testing just now I did find a curious quirk that does point to it being overused resource related though. I ran upscale and diffuse images over 50 times without hitting the issue, but the second I run through the whole process of generating new images, diffuse, and then upscale, that is when I hit the resource limit. 

So if you are running the docker yourself, a workaround to try would be generating images till you get something you want, restart the docker image, and then run diffuse and upscale. It might work for you too. Its a bit of a rough one but it works for now.
